app/views/questionnaire.py

```diff
@@ -280,15 +301,24 @@ def post_block_handler(
             return redirect(next_url)
 
     questionnaire_store_updater.save_answers(form)
-    next_location = path_finder.get_next_location(current_location=current_location)
 
-    if _is_end_of_questionnaire(block, next_location):
+    next_location = path_finder.get_next_location(
+        current_location, get_current_section(user=current_user)
+    )
+
+    if _is_end_of_section(schema.block(next_location.block_id)):
+        current_section = schema.get_section_via_block_id(block_id)
+        questionnaire_store_updater.add_completed_section(current_section['id'])
+
+    if _is_end_of_questionnaire(block):
         return submit_answers(routing_path, schema)
 
+    if not next_location:
+        path_finder.get_next_location(current_location, current_section)
     return redirect(next_location.url())
 
 
-def perform_list_action(
+def perform_list_action(  # pylint: disable=bad-continuation
```

Comment:
I'm assuming this is because there are too many lines? Maybe disable this rule in the config, given that we have black?

Reply:
This can go now